server/container_create_linux.go

```diff
@@ -906,6 +913,95 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 	return ociContainer, nil
 }
 
+func setContainerTimeZone(ctx context.Context, tz, mountPoint, etc, mountLabel string, options []string, ociContainer *oci.Container, ctr ctrfactory.Container) error {
+	var err error
+	if tz == "" {
+		return nil
+	}
+
+	timezonePath := filepath.Join("/usr/share/zoneinfo", tz)
+	if tz == "local" {
+		timezonePath, err = filepath.EvalSymlinks("/etc/localtime")
+		if err != nil {
+			return fmt.Errorf("finding local timezone for container %s: %w", ociContainer.ID(), err)
+		}
+	}
+
+	etcFd, err := unix.Open(etc, unix.O_RDONLY|unix.O_PATH, 0)
+	if err != nil {
+		return fmt.Errorf("open /etc in the container: %w", err)
+	}
+	defer unix.Close(etcFd)
+
+	// Make sure to remove any existing localtime file in the container to not create invalid links
+	err = unix.Unlinkat(etcFd, "localtime", 0)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("removing /etc/localtime: %w", err)
+	}
+
+	hostPath, err := securejoin.SecureJoin(mountPoint, timezonePath)
+	if err != nil {
+		return fmt.Errorf("resolve zoneinfo path in the container: %w", err)
+	}
+
+	if _, err := os.Stat(hostPath); err != nil {
+		// File does not exist, which means tzdata is not installed in the container.
+		// Create /etc/localtime as a copy from the host.
+		log.Debugf(ctx, "Timezone %s does not exist in the container, create our own copy from the host", timezonePath)
+		localtimePath, err := copyTimezoneFile(mountLabel, timezonePath, ociContainer)
+		if err != nil {
+			return fmt.Errorf("setting timezone for container %s: %w", ociContainer.ID(), err)
+		}
+
+		ctr.SpecAddMount(rspec.Mount{
+			Destination: "/etc/localtime",
```

Comment:
You pass in `etc` as a variable here but then you use a hardcoded value of `/etc/localtime`.

Reply:
so the `etc` [value](https://github.com/cri-o/cri-o/pull/7589/files#diff-1e0877112035840d3e2f073c693665f8daffa241a440e54a866ad7382b510ea2R855) passed to this function is different than the one used here.